packages/web-wallet/src/components/CreateTokenDialog.tsx

```diff
@@ -0,0 +1,561 @@
+import React, { useState, useMemo } from 'react';
+import { X, Loader2, AlertCircle, CheckCircle, Copy, Info } from 'lucide-react';
+import { useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { z } from 'zod';
+import { useWallet } from '../contexts/WalletContext';
+import { useInvokeSnap } from '@hathor/snap-utils';
+import helpers from '@hathor/wallet-lib/lib/utils/helpers';
+import { formatHTRAmount } from '../utils/hathor';
+import { readOnlyWalletService } from '../services/ReadOnlyWalletService';
+import { getAddressForMode } from '../utils/addressMode';
+import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from './ui/select';
+
+interface CreateTokenDialogProps {
+  isOpen: boolean;
+  onClose: () => void;
+}
+
+// Zod schema for token creation form
+const createTokenSchema = z.object({
+  name: z
+    .string()
+    .min(1, 'Token name is required')
+    .max(30, 'Token name must be 30 characters or less'),
+  symbol: z
+    .string()
+    .min(2, 'Symbol must be at least 2 characters')
+    .max(5, 'Symbol must be 5 characters or less')
+    .regex(/^[A-Z0-9]+$/, 'Symbol must contain only uppercase letters and numbers')
+    .transform((val) => val.toUpperCase()),
+  amount: z
+    .string()
+    .min(1, 'Amount is required')
+    .regex(/^\d+$/, 'Amount must be a whole number')
+    .refine((val) => {
+      const num = parseInt(val, 10);
+      return num > 0;
+    }, 'Amount must be greater than 0')
+    .refine((val) => {
+      const num = parseInt(val, 10);
+      return num <= Number.MAX_SAFE_INTEGER;
+    }, 'Amount is too large'),
+  tokenType: z.enum(['deposit', 'fee']),
```

Comment:
Oh, we are already accounting for fee based tokens?
We should update all references of the config strings to include the token info version